src/engine/framework/System.cpp

```diff
@@ -748,6 +749,11 @@ ALIGN_STACK_FOR_MINGW int main(int argc, char** argv)
 					Log::Notice(err.what());
 				}
 				Application::OnDrop(err.is_error(), err.what());
+
+				if (Sys::shutdownOnDrop.Get()) {
+					Sys::Shutdown(false, "Shutdown on drop");
+					Sys::OSExit(0);
```

Comment:
It might be useful to make the return code different based on `err.is_error()`

Reply:
See my new commit.